src/helpers/issue.ts

```diff
@@ -314,6 +314,48 @@ export const removeAssignees = async (issue_number: number, assignees: string[])
   }
 };
 
+export const checkUserPermissionForRepoAndOrg = async (username: string, context: Context): Promise<boolean> => {
+  const permissionForRepo = await checkUserPermissionForRepo(username, context);
+  const permissionForOrg = await checkUserPermissionForOrg(username, context);
+
+  return permissionForOrg || permissionForRepo;
+};
+
+export const checkUserPermissionForRepo = async (username: string, context: Context): Promise<boolean> => {
+  const logger = getLogger();
+  const payload = context.payload as Payload;
+
+  try {
+    const res = await context.octokit.rest.repos.checkCollaborator({
+      owner: payload.repository.owner.login,
+      repo: payload.repository.name,
+      username,
+    });
+
+    return res.status === 204;
+  } catch (e: unknown) {
+    logger.debug(`Checking if user permisson for repo failed!, reason: ${e}`);
+    return false;
+  }
+};
+
+export const checkUserPermissionForOrg = async (username: string, context: Context): Promise<boolean> => {
+  const logger = getLogger();
+  const payload = context.payload as Payload;
+  if (!payload.organization) return false;
+
+  try {
+    await context.octokit.rest.orgs.checkMembershipForUser({
+      org: payload.organization.login,
+      username,
+    });
+    return true;
```

Comment:
you have to check for status `204` according to [documentation](https://docs.github.com/en/rest/orgs/members?apiVersion=2022-11-28#check-organization-membership-for-a-user)

Reply:
seems it only returns 302 status. type error occours when check status 204.
That's why I didn't check it.
https://github.com/octokit/rest.js/issues/188

Reply:
Ideally, we should check the response status and return true if `status` is 204.
It will make arbitrary contributors understand what the success status is, what the failure status is, etc